### PR TITLE
feat: future goals

### LINKED ISSUE #2350: [FEATURE] Future milestones to have a dotted line
### Description

We have milestones at the moment, but these are in the past, it would be great to have goals - these would need to be displayed slightly differently

Maybe a flag that is `goal: true`

Current data...

```json
"milestones": [
    {
      "title": "Started Freelancing",
      "date": "May 2010",
      "icon": "dollar",
      "color": "grey",
      "description": "Started freelancing again",
      "url": "https://www.eddiejaoude.io/"
    }
]
```

We could add a flag...
```json
"milestones": [
    {
      "isGoal": true,
      "title": "Started Freelancing",
      "date": "May 2010",
      "icon": "dollar",
      "color": "grey",
      "description": "Started freelancing again",
      "url": "https://www.eddiejaoude.io/"
    }
]
```

Currently milestones are displayed like this, but we could have a dotted line down to another group of cards and these could be slightly faded, to show they have not been achieved yet

Screenshot 2022-12-11 at 11 34 01

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ManishBisht777**
## Fixes Issue #2350 

Closes #2350 

## Changes proposed

User can add future goals to their profile

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

Screenshot from 2022-12-12 17-03-28

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/2383"></a>

### CODE REVIEW

**Review by ManishBisht777 — COMMENTED** [review]
we can have the future goals text conditionally rendered by computing the goals beforehand using `useEffect` and `useState` hook

File: components/user/UserGoal.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            </div>
+            <p className="text-sm text-gray-500">
+              <ReactMarkdown>{goal.description}</ReactMarkdown>
+            </p>
+          </div>
+        </div>
+      </li>
+    </a>
+  );
+}
```
Comment (eddiejaoude) [inline components/user/UserGoal.js:36]: Can we not use the existing component rather than create a new one?

Comment (ManishBisht777) [reply components/user/UserGoal.js:36]: yes we can 
1)  by passing `isgoal` to the existing component but we need to apply the style then conditionally ( opacity )
2) 2 divs rendered according to `isgoal`

but  then I thought if we have some other plans for the goals section then keeping it in another component would be good

Comment (eddiejaoude) [reply components/user/UserGoal.js:36]: I think there will not be anymore functionality, and we can always split it out if needed later - so I think 1 component for now would be best 👍

**Review by ManishBisht777 — COMMENTED** [review]
@eddiejaoude Merged UserGoal component into a userMilestone component

**Review by schmelto — CHANGES_REQUESTED** [review]
Hey it seems there is a merge conflict could you resolve it?

**Review by ManishBisht777 — COMMENTED** [review]
Resolved the merge conflicts

File: components/user/UserMilestone.js [hunk 2]
```diff
@@ -1,8 +1,38 @@
 import Icon from "../Icon";
 import { ReactMarkdown } from "react-markdown/lib/react-markdown";
 
-export default function UserMilestone({ milestone }) {
-  return (
+export default function UserMilestone({ milestone, isGoal }) {
+  return isGoal ? (
+    <a
+      href={milestone.url ? milestone.url : "/"}
```
Comment (eddiejaoude) [inline components/user/UserMilestone.js:7]: Not sure I understand the reason for the `/`?

File: components/user/UserMilestone.js [hunk 3]
```diff
@@ -1,8 +1,38 @@
...
+              </p>
+            </div>
+            <ReactMarkdown className="text-sm text-gray-500">
+              {milestone.description}
+            </ReactMarkdown>
+          </div>
+        </div>
+      </li>
+    </a>
+  ) : (
```
Comment (eddiejaoude) [inline components/user/UserMilestone.js:35]: Can they not use the same code?

**Pull request merged**